politeiawww/cmd/politeiawww_dbutil/politeiawww_dbutil.go

```diff
@@ -330,83 +543,256 @@ func stubUsersAction() error {
+			if err != nil {
+				return fmt.Errorf("set paywall index: %v", err)
+			}
+		default:
+			// User record
+			u, err := user.DecodeUser(value)
+			if err != nil {
+				return fmt.Errorf("decode user '%v': %v",
+					value, err)
+			}
+			err = cdb.InsertUser(*u)
+			if err != nil {
+				return fmt.Errorf("migrate user '%v': %v",
+					u.ID, err)
+			}
+			userCount++
 		}
+	}
 
-		i++
+	// Make sure the migration went ok.
+	if userCount == 0 {
+		return fmt.Errorf("no users found in leveldb")
+	}
+
+	if paywallIndex == 0 {
+		return fmt.Errorf("paywall address index not found")
 	}
 
+	fmt.Printf("Users migrated : %v\n", userCount)
+	fmt.Printf("Paywall index  : %v\n", paywallIndex)
 	fmt.Printf("Done!\n")
+
+	iter.Release()
+	return iter.Error()
+}
+
+func cmdCreateKey() error {
+	path := defaultEncryptionKey
+	args := flag.Args()
+	if len(args) > 0 {
+		path = util.CleanAndExpandPath(args[0])
+	}
+
+	// Don't allow overwriting an existing key
+	_, err := os.Stat(path)
+	if err == nil {
+		return fmt.Errorf("file already exists; cannot "+
+			"overwrite %v", path)
+	}
+
+	// Create key
+	err = util.NewEncryptionKey(path)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Encryption key saved to: %v\n", path)
+	return nil
+}
+
+func validateCockroachParams() error {
+	// Validate host
+	_, err := url.Parse(*host)
+	if err != nil {
+		return fmt.Errorf("parse host '%v': %v",
+			*host, err)
+	}
+
+	// Validate root cert
+	b, err := ioutil.ReadFile(*rootCert)
+	if err != nil {
+		return fmt.Errorf("read rootcert: %v", err)
+	}
+
+	block, _ := pem.Decode(b)
+	_, err = x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return fmt.Errorf("parse rootcert: %v", err)
+	}
+
+	// Validate client key pair
+	_, err = tls.LoadX509KeyPair(*clientCert, *clientKey)
+	if err != nil {
+		return fmt.Errorf("load key pair clientcert and "+
+			"clientkey: %v", err)
+	}
+
+	// Validate encryption key
+	_, err = util.LoadEncryptionKey(*encryptionKey)
+	if err != nil {
+		return fmt.Errorf("load encryption key: %v", err)
+	}
+
 	return nil
 }
 
 func _main() error {
 	flag.Parse()
 
-	var net string
+	*dataDir = util.CleanAndExpandPath(*dataDir)
+	*rootCert = util.CleanAndExpandPath(*rootCert)
+	*clientCert = util.CleanAndExpandPath(*clientCert)
+	*clientKey = util.CleanAndExpandPath(*clientKey)
+	*encryptionKey = util.CleanAndExpandPath(*encryptionKey)
+
 	if *testnet {
-		net = chaincfg.TestNet3Params.Name
+		network = chaincfg.TestNet3Params.Name
 	} else {
-		net = chaincfg.MainNetParams.Name
+		network = chaincfg.MainNetParams.Name
 	}
 
-	dbDir = filepath.Join(*dataDir, net, localdb.UserdbPath)
-	fmt.Printf("Database: %v\n", dbDir)
+	// Validate database selection
+	if *level && *cockroach {
+		return fmt.Errorf("database choice cannot be both " +
+			"--leveldb and --cockroachdb")
+	}
 
-	if _, err := os.Stat(dbDir); os.IsNotExist(err) {
-		return fmt.Errorf("database directory does not exist: %v",
-			dbDir)
+	switch {
```

Comment:
This switch case statement seems to alternate between levelDB and LocalDB. Is this the correct behavior? I wasn't able to run `politeiawww_dbutil dump` using the -localdb flag if so

Reply:
`-localdb` was a type.  Fixed it.  Good catch.